### PR TITLE
Fix Hlabel and add medium test case

### PR DESCRIPTION
**Pull request by sungmanc**
### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

**Review by sovrasov — DISMISSED** [review]

File: src/otx/algo/classification/metrics/hlabel_accuracy.py [hunk 1]
```diff
@@ -72,9 +72,10 @@ def update(self, preds: torch.Tensor, target: torch.Tensor) -> None:
             target_multiclass = target[:, head_idx]
             multiclass_mask = target_multiclass > 0
 
+            # Torchmetric should be located at the same device with tensors.
+            self._metric_to_device(self.multiclass_head_accuracy[head_idx], preds.device)
```
Comment (vinnamkim) [inline src/otx/algo/classification/metrics/hlabel_accuracy.py]: Removing this and can we make an overridden function for this?
https://github.com/Lightning-AI/torchmetrics/blob/7d23f974cde38ee233b71dfda006e52834604ee6/src/torchmetrics/metric.py#L776
For example,

```python
class HLabelAccuracy(Metric):
    ...
    def _apply(self, fn: Callable, exclude_state: Sequence[str] = "") -> Module:
        self.multiclass_head_accuracy = [acc._apply(fn, exclude_state) for acc in self.multiclass_head_accuracy]
        self.multilabel_accuracy = self.multilabel_accuracy._apply(fn, exclude_state)
        return self
```

Comment (sungmanc) [reply src/otx/algo/classification/metrics/hlabel_accuracy.py]: Thanks for the nice comment, done https://github.com/openvinotoolkit/training_extensions/pull/2804/commits/68abcedffe38e68b3423d28ac5f32e975c577117

**Comment by sungmanc** [comment]
@sovrasov , thank you for the comment about the missing information that Model API wants.

Is it enough? Maybe I'm wrong. please check. When I checked the OTX1.5, multi-label information wasn't included in the `label_tree_edges` information.  
https://github.com/openvinotoolkit/training_extensions/pull/2804/commits/4d30196abe63c5d4bc317e30ab5f6f9e8d0361d5

In the case of tests/assets/hlabel_classification, HLabelInfo will be
image

Label
image

**Review by vinnamkim — APPROVED** [review]

**Comment by sovrasov** [comment]
> [4d30196](https://github.com/openvinotoolkit/training_extensions/commit/4d30196abe63c5d4bc317e30ab5f6f9e8d0361d5)

As I said, a list of pairs (child, parent) is missing: https://github.com/openvinotoolkit/training_extensions/blob/develop/src/otx/algorithms/classification/utils/cls_utils.py#L120
This wasn't included in hierarchical_info itself, but in hierarchical config for MAPI. Without that we can't use MAPI properly. Now, I propose just to add this to HLabelInfo to avoid new structures.

**Comment by sungmanc** [comment]
> > [4d30196](https://github.com/openvinotoolkit/training_extensions/commit/4d30196abe63c5d4bc317e30ab5f6f9e8d0361d5)
> 
> As I said, a list of pairs (child, parent) is missing: https://github.com/openvinotoolkit/training_extensions/blob/develop/src/otx/algorithms/classification/utils/cls_utils.py#L120 This wasn't included in hierarchical_info itself, but in hierarchical config for MAPI. Without that we can't use MAPI properly. Now, I propose just to add this to HLabelInfo to avoid new structures.

Could you check again? I added that information to the HLabelInfo. Do you mean the other information?
https://github.com/openvinotoolkit/training_extensions/blob/4d30196abe63c5d4bc317e30ab5f6f9e8d0361d5/src/otx/core/data/entity/classification.py#L285

**Comment by sovrasov** [comment]
> > > [4d30196](https://github.com/openvinotoolkit/training_extensions/commit/4d30196abe63c5d4bc317e30ab5f6f9e8d0361d5)
> > 
> > 
> > As I said, a list of pairs (child, parent) is missing: https://github.com/openvinotoolkit/training_extensions/blob/develop/src/otx/algorithms/classification/utils/cls_utils.py#L120 This wasn't included in hierarchical_info itself, but in hierarchical config for MAPI. Without that we can't use MAPI properly. Now, I propose just to add this to HLabelInfo to avoid new structures.
> 
> Could you check again? I added that information to the HLabelInfo. Do you mean the other information?
> 
> https://github.com/openvinotoolkit/training_extensions/blob/4d30196abe63c5d4bc317e30ab5f6f9e8d0361d5/src/otx/core/data/entity/classification.py#L285

Thanks, that would be enough. `[label, label_group.name]` looks a bit suspicious to me, it would work only if the parent label's name always equals to the name of the label group. I'm not sure here, because not familiar with datumaro's hierarchical data representation.

**Review by sovrasov — APPROVED** [review]

**Review by harimkang — APPROVED** [review]

**Pull request merged**